app/services/hyrax/resource_characterizer.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # Characterizes a Valkyrie based FileSet
+  class ResourceCharacterizer
+    ##
+    # @!attribute [rw] source
+    #   @return [#characterize]
+    attr_accessor :source
+
+    ##
+    # @param source the object to characterize
+    def initialize(source:)
+      @source = source
+    end
+
+    ##
+    # @return [void]
+    #
+    # @raise [RuntimeError] if FileSet is missing the characterization_proxy
+    def characterize
+      Hydra::Works::CharacterizationService.run(characterization_proxy, filepath)
```

Comment:
what do you think about injecting this dependency?

```ruby
##
# @param service [#run]
def characterize(service: Hydra::Works::CharacterizationService)
  service.run(characterization_proxy, filepath)
  ...
end
```

in this case, we could inject a spy service in the test suite and check its arguments, instead of hard-stubbing the class method.